doc/reqparse

```diff
@@ -0,0 +1,153 @@
+                elif state == State.REQ_HEADER_PRE:
+                    error(filename, i, 1, "unexpected \"---\",\n"
+                            "expected another line of req header marker")
+                elif state == State.REQ_HEADER:
+                    parse_header(current_entity, filename, header_line)
+                    state = State.REQ_TEXT
+                elif state == State.REQ_TEXT:
+                    error(filename, i, 1, "unexpected \"---\",\n"
+                            "expected markdown or next req header marker")
+                else:
+                    error(filename, i, 1, "something went terribly wrong")
+            elif l.strip() == "-"*80:
+                if state == State.BEGIN:
+                    error(filename, i, 1, "unexpected req header marker,\n"
+                            "expected group header marker")
+                elif state == State.GROUP_HEADER:
+                    error(filename, i, 1, "unexpected req header marker,\n"
+                            "expected yaml or group header end marker")
+                elif state == State.GROUP_TEXT:
+                    group = current_entity
+                    current_entity = None
+                    state = State.REQ_HEADER_PRE
+                elif state == State.REQ_HEADER_PRE:
+                    current_entity = Entity()
+                    state = State.REQ_HEADER
+                    header_line = i
+                elif state == State.REQ_HEADER:
+                    error(filename, i, 1, "unexpected req header marker,\n"
+                            "expected markdown or req header end marker")
+                elif state == State.REQ_TEXT:
+                    reqs.append(current_entity)
+                    current_entity = None
+                    state = State.REQ_HEADER_PRE
+                else:
+                    error(filename, i, 1, "something went terribly wrong")
+            else:
+                if state == State.BEGIN:
+                    error(filename, i, 1, "unexpected character, expected group header")
+                elif state == State.GROUP_HEADER:
+                    current_entity.header_text += l
+                elif state == State.GROUP_TEXT:
+                    current_entity.text += l
+                elif state == State.REQ_HEADER_PRE:
+                    error(filename, i, 1, "expected another line of req header marker")
+                elif state == State.REQ_HEADER:
+                    current_entity.header_text += l
+                elif state == State.REQ_TEXT:
+                    current_entity.text += l
+                else:
+                    error(filename, i, 1, "something went terribly wrong")
+    if state == State.REQ_TEXT:
+            reqs.append(current_entity)
+    return (group, reqs)
+
+
+parser = argparse.ArgumentParser(prog="reqparse")
+parser.add_argument(
+    "-o", "--output", action="store",
+    help="Output file"
+)
```

Comment:
I know the user should explicitly specify output file, but what do You think about adding suffix dependent on output format?

Reply:
This is not Linux-style behavior.